falass/sld.py

```diff
@@ -37,43 +37,44 @@ def get_sld_profile(self):
-                for j in range(0, number_of_bins):
-                    build_sld.append(dataformat.SLDPro(self.assigned_job.layer_thickness, 0, 0))
-                self.sld_profile.append(build_sld)
+
+        # create mask of which frames to analyse
+        time_mask = np.array([True if t in self.assigned_job.times else False
+                              for t in self.assigned_job.files.times], dtype=bool)
+
+        u = self.assigned_job.files.u
+
         k = 0
-        for i in range(0, len(self.assigned_job.files.atoms)):
-            if job.check_array(self.assigned_job.times, self.assigned_job.files.times[i]):
-                z_cut = self.assigned_job.files.cell[i][2] - self.assigned_job.cut_off_size
-                number_of_bins = int(z_cut / self.assigned_job.layer_thickness)
-                for j in range(0, len(self.assigned_job.files.atoms[i])):
-                    if self.assigned_job.files.atoms[i][j].zpos < number_of_bins * self.assigned_job.layer_thickness:
-                        bin_choose = int(self.assigned_job.files.atoms[i][j].zpos / self.assigned_job.layer_thickness)
-                        scatlen_to_add = get_scatlen(self.assigned_job.files.atoms[i][j].atom,
-                                                     self.assigned_job.files.scat_lens)
-                        self.sld_profile[i][bin_choose].real += scatlen_to_add[0]
-                        self.sld_profile[i][bin_choose].imag += scatlen_to_add[1]
-                    k += 1
-                    prog_new = np.floor(k / (len(self.assigned_job.files.atoms[i]) * 
-                                             len(self.assigned_job.files.atoms)) * 100)
-                    if prog_new > prog + 9:
-                        prog = prog_new
-                        print("[{} {} % ]".format('#' * int(prog / 10), int(prog / 10) * 10))
-        for i in range(0, len(self.assigned_job.files.atoms)):
-            if job.check_array(self.assigned_job.times, self.assigned_job.files.times[i]) is True:
-                z_cut = self.assigned_job.files.cell[i][2] - self.assigned_job.cut_off_size
-                number_of_bins = int(z_cut / self.assigned_job.layer_thickness)
-                for j in range(0, number_of_bins):
-                    self.sld_profile[i][j].real = self.sld_profile[i][j].real / (self.assigned_job.files.cell[i][0] *
-                                                                                 self.assigned_job.files.cell[i][1] *
-                                                                                 self.assigned_job.layer_thickness)
-                    self.sld_profile[i][j].imag = self.sld_profile[i][j].imag / (self.assigned_job.files.cell[i][0] *
-                                                                                 self.assigned_job.files.cell[i][1] *
-                                                                                 self.assigned_job.layer_thickness)
+        for ts in u.trajectory[time_mask]:
```

Comment:
`u` is the class which represents the PDB file you loaded.

`u.trajectory` represents the concept of the trajectory that the loaded PDB has
`u.atoms` gives information on atoms for the currently loaded frame

This loop is iterating through the frames you want to analyse.  Each time this iterates in the loop the new positions and cell dimensions are loaded into `u.atoms.positions` and `u.dimensions` respectively.

For example this loop prints the position of the first atom:
```python
for ts in u.trajectory:
    print(u.atoms[0].position)
```